src/lib/components/BlogPage.svelte

```diff
@@ -0,0 +1,198 @@
+  }
+
+  $: sortedAndFilteredBlogs = blogListContents
+    ? blogListContents
+        .filter((blog) => filterTag === "All" || blog.tags.includes(filterTag))
+        .sort((a, b) => {
+          switch (sortKey) {
+            case true:
+              return a.publishedAt.localeCompare(b.publishedAt);
+            case false:
+              return b.publishedAt.localeCompare(a.publishedAt);
+            default:
+              return 0;
+          }
+        })
+        .slice(start, end)
+    : [];
+
+  $: morePage = blogListContents ? blogListContents.length > end : false;
+
+  let selectedIcon = "";
+
+  $: {
+    selectedIcon = "";
+    for (const key in tags) {
+      if (tags[key].title === filterTag) {
+        selectedIcon = tags[key].icon;
+        break;
+      }
+    }
+  }
+
+  const items = getEntries(tags).map(([value, { title, icon }]) => ({
+    value,
+    label: title,
+    icon,
+  }));
+</script>
+
+{#if more}
+  <div id="sort">
+    <Select.Root {items} selected={{ value: filterTag }}>
+      <Select.Trigger title="タグを選択する">
+        <div class="icon">
+          <Icon icon={selectedIcon} height={23} />
+          <p>{filterTag}</p>
+        </div>
+      </Select.Trigger>
+      <Select.Content
+        class="select-content"
+        sameWidth={false}
+        sideOffset={8}
+        transition={slide}
+        transitionConfig={{ duration: 300 }}
+      >
+        {#each items as { label, icon, value }}
+          <Select.Item
+            class="select-item"
+            {value}
+            {label}
+            on:click={() => {
+              filterTag = label;
+            }}
+          >
+            <Icon {icon} />
+            <p>{label}</p>
+            <div class="icon">
+              <Select.ItemIndicator
+                class="select-item-indicator"
+                asChild={false}
+              >
+                <Icon icon="mdi:check" />
+              </Select.ItemIndicator>
+            </div>
+          </Select.Item>
+        {/each}
+        <Select.Input name="favoriteFruit" />
+      </Select.Content>
+    </Select.Root>
+    <Tag on:click={handlePublishedAtClick}>投稿日</Tag>
+  </div>
+{/if}
+
+<div class="content">
+  {#if sortedAndFilteredBlogs.length === 0}
+    <p id="no-page">ページは存在しません</p>
+  {:else}
+    {#await blogs then value}
+      {#each value.contents as blogData}
+        <BlogLink {blogData} />
+      {/each}
+    {:catch error}
+      <p>Error: {error.message}</p>
+    {/await}
+  {/if}
+</div>
+<div id="next">
+  {#if page > 1 && sortedAndFilteredBlogs.length !== 0}
+    <button
+      on:click={() => {
+        page -= 1;
+        window.scrollTo(0, 0);
+      }}>前のページ</button
+    >
+  {:else}
+    <span></span>
+  {/if}
+
+  {#if more && morePage && sortedAndFilteredBlogs.length !== 0}
+    <button
+      on:click={() => {
+        page += 1;
+        window.scrollTo(0, 0);
+      }}>次のページ</button
+    >
+  {/if}
+</div>
+
+<style lang="scss">
+  #sort {
```

Comment:
idセレクタはどのセレクタより強力だから、
classセレクタで問題ない場合は idセレクタは避けた方がいいかも
(他も同様)
```suggestion
  .sort {
```
※これもやや宗教なので従う必要はない

js で使う場合や
`.container > .box > .... > .text` みたいに深くなりすぎる場合は全然いいと思う